engine/cdn/redis/readwrite.go

```diff
@@ -17,16 +19,48 @@ type ReadWrite struct {
 	UsageKey  string
 }
 
-func (rw *ReadWrite) get(from uint, to uint) ([]string, error) {
-	var res = make([]string, to-from+1)
+type Line struct {
+	NumberString string `json:"-"`
+	Number       int64  `json:"number"`
+	Value        string `json:"value"`
+}
+
+func (l Line) Format(f sdk.CDNReaderFormat) ([]byte, error) {
+	switch f {
+	case sdk.CDNReaderFormatJSON:
+		var err error
+		l.Number, err = strconv.ParseInt(l.NumberString, 10, 64)
+		if err != nil {
+			return nil, sdk.WrapError(err, "cannot parse line number with value %s", l.NumberString)
+		}
+		bs, err := json.Marshal(l)
+		return bs, sdk.WithStack(err)
+	case sdk.CDNReaderFormatText:
+		return []byte(l.Value), nil
+	}
+	return nil, sdk.WithStack(fmt.Errorf("invalid given reader format '%s'", f))
+}
+
+func (rw *ReadWrite) get(from uint, to uint) ([]Line, error) {
+	res := make([]string, to-from+1)
 	if err := rw.Store.ScoredSetScan(context.Background(), cache.Key(rw.PrefixKey, rw.ItemID), float64(from), float64(to), &res); err != nil {
-		return res, err
+		return nil, err
 	}
+	ls := make([]Line, len(res))
 	for i := range res {
-		res[i] = strings.TrimFunc(res[i], unicode.IsNumber)
-		res[i] = strings.TrimPrefix(res[i], "#")
+		tmp := strings.SplitN(res[i], "#", 2)
```

Comment:
why did you change this ?
the previous implementation wasn't error prone.
Here is there is any weird data in redis, we are unable to read the whole log :(

Reply:
This was changed because we need to get the line number from redis, I edit my first version to get the value from redis score instead of parsing the log value.